---
int32 error_code
string error_description